js/src/dropdown.js

```diff
@@ -405,11 +404,17 @@ class Dropdown extends BaseComponent {
         continue
       }
 
-      if (event && ((event.type === 'click' &&
-          /input|textarea/i.test(event.target.tagName)) ||
-          (event.type === 'keyup' && event.key === TAB_KEY)) &&
-          dropdownMenu.contains(event.target)) {
-        continue
+      if (event) {
+        if (context._element === SelectorEngine.parents(event.target, SELECTOR_DATA_TOGGLE)[0]) {
```

Comment:
@rohit2sharma95 I also need to compare `context._element` with `event.target` to ensure clicks on the button itself toggle the menu, but that increases the complexity to 21, causing eslint to fail. Any ideas?

Reply:
It's a warning, although it'd be nice to not have it. We should try to move some logic outside of `clearMenus()` at some point

Reply:
I switched to `event.composedPath().includes(context._element)` instead. `composedPath` seems to be pretty standard in modern browsers (https://developer.mozilla.org/en-US/docs/Web/API/Event/composedPath), but let me know if you'd prefer I switch back.

Reply:
Not against this if it's supported but perhaps we should make that change in a separate PR in case other compononents can use it? https://caniuse.com/?search=composedPath

Reply:
The alternative would be something like:

```js
if (context._element === event.target || context._element === SelectorEngine.parents(event.target, SELECTOR_DATA_TOGGLE)[0] || dropdownForm === event.target || dropdownForm === SelectorEngine.parents(event.target, SELECTOR_FORM_CHILD)[0]) {
    ...
}
```

Could we maybe merge this as is and then open another issue addressing using `composedPath` in other places where it would make sense?